src/env_var.zig

```diff
@@ -356,7 +356,7 @@ const kind = struct {
                     /// Note: Most values are considered truthy, except for "", "0", "false", "no",
                     /// and "off".
                     truthy_cast,
-                } = .panic,
+                } = if (bun.Environment.isDebug) .panic else .default_fallback,
```

Comment:
If users pass environment variables that they expect to have some meaning, we must communicate to them that we're disregarding what they passed.
There was an example of a misuse of RedisClient where a user passed a malformed URL. We then fell back to localhost:6379 and they were very confused and filed a ticket.
Silent failures cost real dollars and if we want people to treat bun seriously, we need to be very verbose about user errors.

This PR changes the semantics of user failures to silent failures and that's going to cost someone real money.